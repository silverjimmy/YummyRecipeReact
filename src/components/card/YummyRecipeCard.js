import React from 'react';
import { Link } from "react-router";
import "./card.css";


const YummyRecipeCard = (props) => (
  <section className="tiles">
    {
      props.categories.map((category, index) => (
        <div key={index} >
            <article className="style10">
              <span className="image">
                <img src={ process.env.PUBLIC_URL + "/images/pic02.jpg"} alt="" />
              </span>
              <Link onClick={(event) => props.redirect(event, category.id, category.title)}>
                <h2>{ category.title }</h2>
                <div className="content">
                  <p>{ category.description }</p>

                </div>
              </Link>
            </article>
          </div>
      ))
    }
  </section>
);

export default YummyRecipeCard;
