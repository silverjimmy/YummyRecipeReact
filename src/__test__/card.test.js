
import React from 'react';
import ReactDOM from 'react-dom';
import { render,  configure,shallow } from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import expect from "expect";
import Adapter from 'enzyme-adapter-react-15';
import YummyRecipeCard from '../components/card/YummyRecipeCard';
import LocalStorageMock from '../setupTests';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';


configure({ adapter: new Adapter() });

let match = {
    params : '1'
}

const setUp = () => {
    const props = {
        match
    };
    return shallow(<YummyRecipeCard {...props} />);
};

describe('ViewRecipe', () => {
    const wrapper = shallow( <YummyRecipeCard /> );
    it('matches  snapshot', () => {
            const viewRecipe = setUp;
           expect(shallowToJson(viewRecipe)).toMatchSnapshot();
        });    
        it('renders div jsx elements', () => {
            expect(wrapper.find("div")).toHaveLength(0);        
            });

        it('renders span jsx elements', () => {
            expect(wrapper.find("span")).toHaveLength(0);        
            });
        
        it('renders article jsx elements', () => {
            expect(wrapper.exists("article"));        
            });
        it('should render <div> without throwing an error', () => {
            expect(wrapper.exists(<div/>))
        });

        it('should render <div> without throwing an error', () => {
            expect(wrapper.exists(<h1 />))
        });
        it('should render <div> without throwing an error', () => {
            expect(wrapper.exists(<article />))
        });

})
