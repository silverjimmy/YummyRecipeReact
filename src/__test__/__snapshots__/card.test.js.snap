// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRecipe matches  snapshot 1`] = `null`;
